src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -522,18 +567,37 @@ def distribute_shares(shares: List[ShareTensor], session: Session) -> List:
         Returns:
             List of ShareTensor.
 
+        Raises:
+            TypeError: when Datatype of shares is invalid.
+
         """
-        parties = session.parties
+        if not isinstance(shares, list):
+            raise TypeError("Shares to be distributed should be a list of shares")
+
+        if len(shares) == 0:
```

Comment:
MAYBE: Remove this, since this case would be already covered by the bellow code, right?